cloud/amazon: TestS3BucketDoesNotExist failed

cloud/amazon.TestS3BucketDoesNotExist [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/12719968?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/12719968?buildTab=artifacts#/) on release-23.1 @ [9730331acb6fa7b197dc7ac1e4b561ec84df26bb](https://github.com/cockroachdb/cockroach/commits/9730331acb6fa7b197dc7ac1e4b561ec84df26bb):


```
=== RUN   TestS3BucketDoesNotExist
    s3_storage_test.go:457: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/cloud/amazon/s3_storage_test.go:457
        	Error:      	Should be true
        	Test:       	TestS3BucketDoesNotExist
--- FAIL: TestS3BucketDoesNotExist (0.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestS3BucketDoesNotExist.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33601

--- a/pkg/cloud/amazon/s3_storage_test.go
+++ b/pkg/cloud/amazon/s3_storage_test.go
@@ -515,7 +515,8 @@ func TestS3BucketDoesNotExist(t *testing.T) {
 	q.Add(cloud.AuthParam, cloud.AuthParamImplicit)
 	q.Add(S3RegionParam, "us-east-1")
 
-	bucket := "invalid-bucket"
+	// Very long, very random bucket name, that hopefully nobody will ever create.
+	bucket := "VIBK1H88MOJ665V2RAPVH6X3TWUS0HCWTW5A27AFPPLHMABKH7X445K86K1BP2"
 	u := url.URL{
 		Scheme:   "s3",
 		Host:     bucket,

